File: structural/adapter/adapter.py
class Movable:
    def get_speed(self):
        pass


class BugattiVeyron(Movable):
    def get_speed(self):
        return 268  # // in Mph


class MovableAdapter:
    def get_speed(self):
        pass


def convert_mph_to_kmph(mph):
    return mph * 1.60934


class MovableAdapterImpl(MovableAdapter):
    def __init__(self, car):
        super().__init__()
        self.car = car

    def get_speed(self):
        return convert_mph_to_kmph(self.car.get_speed())


bugattiVeyron = BugattiVeyron()
bugattiVeyronAdapter = MovableAdapterImpl(bugattiVeyron)
getBugattiVeyronSpeedInKmph = bugattiVeyronAdapter.get_speed()

File: structural/adapter/adapter.js
class Movable {
  getSpeed() {}  // returns speed in Kmph
}

class BugattiVeyron extends Movable {
  getSpeed() {
    return 268; // in Mph
  }
}

class MovableAdapter {
  getSpeed() {} // returns speed in Kmph
}

class MovableAdapterImpl extends MovableAdapter {
  constructor(car) {
    super();
    this.car = car;
  }

  getSpeed() {
    return this.convertMPHtoKMPH(this.car.getSpeed());
  }

  convertMPHtoKMPH(mph) {
    return mph * 1.60934;
  }
}

const bugattiVeyron = new BugattiVeyron();
const bugattiVeyronAdapter = new MovableAdapterImpl(bugattiVeyron);
const getBugattiVeyronSpeedInKmph = bugattiVeyronAdapter.getSpeed();
